// @remotion/licensing/dist/register-usage-point.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.registerUsageEvent = exports.HOST = void 0;
exports.HOST = 'https://www.remotion.pro';
const registerUsageEvent = async ({ apiKey, host, succeeded, event, }) => {
    const abortController = new AbortController();
    const timeout = setTimeout(() => {
        abortController.abort();
    }, 10000);
    try {
        const res = await fetch(`${exports.HOST}/api/track/register-usage-point`, {
            method: 'POST',
            body: JSON.stringify({
                event,
                apiKey,
                host,
                succeeded,
            }),
            headers: {
                'Content-Type': 'application/json',
            },
            signal: abortController.signal,
        });
        clearTimeout(timeout);
        const json = (await res.json());
        if (json.success) {
            return {
                billable: json.billable,
                classification: json.classification,
            };
        }
        if (!res.ok) {
            throw new Error(json.error);
        }
        const read = await res.json();
        return read;
    }
    catch (err) {
        clearTimeout(timeout);
        if (err instanceof Error && err.name === 'AbortError') {
            throw new Error('Request timed out after 10 seconds');
        }
        throw err;
    }
};
exports.registerUsageEvent = registerUsageEvent;
